src/js/libcs/OpSound.js

```diff
@@ -76,24 +96,35 @@ evaluator.playsin$1 = function(args, modifs) {
     let audioCtx = OpSound.getAudioContext();
     let freq = evaluate(args[0]).value.real;
     let line = OpSound.handleLineModif(modifs.line, "0");
-    let amp = OpSound.handleModifs(modifs.amp, 'number', 0.5);
-    let damp = OpSound.handleModifs(modifs.damp, 'number', 0);
-    let stop = OpSound.handleModifs(modifs.stop, 'number', 1);
-    let duration = OpSound.handleModifs(modifs.duration, 'number', stop);
-    let harmonics = OpSound.handleModifs(modifs.harmonics, 'list', [1]);
-    let partials = OpSound.handleModifs(modifs.partials, 'list', [1]);
-    let attack = OpSound.handleModifs(modifs.attack, 'number', 0.01);
-    let release = OpSound.handleModifs(modifs.release, 'number', 0.01);
-    let restart = OpSound.handleModifs(modifs.restart, 'boolean', true);
-    let phaseshift = OpSound.handleModifs(modifs.phaseshift, 'number', 0);
-    let pan = OpSound.handleModifs(modifs.pan, 'number', 0);
-
+    let amp = OpSound.handleModif(modifs.amp, 'number', 0.5);
+    let damp = OpSound.handleModif(modifs.damp, 'number', 0);
+    let stop = OpSound.handleModif(modifs.stop, 'number', 1);
+    let duration = OpSound.handleModif(modifs.duration, 'number', stop);
+    let harmonics = OpSound.handleModif(modifs.harmonics, 'list', [1]);
+    let partials = OpSound.handleModif(modifs.partials, 'list', [1]);
+    let attack = OpSound.handleModif(modifs.attack, 'number', 0.01);
+    let release = OpSound.handleModif(modifs.release, 'number', 0.01);
+    let restart = OpSound.handleModif(modifs.restart, 'boolean', true);
+    let phaseshift = OpSound.handlePhaseshiftModif(modifs.phaseshift, 'list', harmonics.length);
+    let pan = OpSound.handleModif(modifs.pan, 'number', 0);
     let noPartialsGiven = harmonics.length > partials.length;
 
+
+    if (!noPartialsGiven) {
+        for (let i = 0, j = 0; i < harmonics.length; i++) {
+            if (Math.abs(partials[i] - 1) < 1e-10) j++;
```

Comment:
`noPartialsGiven = noPartialsGiven & (Math.abs(partials[i] - 1) < 1e-10)`
would be more simple and `j` could be avoided.

(and partials[i] might not exist before `if (harmonics.length > partials.length) { partials = Array(harmonics.length).fill(1); }`)